pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -124,6 +141,99 @@ func (s secretResource) createSecret(name, label string, identities []byte) erro
+		if ownerRef := metav1.GetControllerOf(keystoreSecret); ownerRef != nil && ownerRef.UID != s.infinispan.UID {
+			return fmt.Errorf("Unable to SetControllerReference for secret %s as it's already owned  by %s:%s:%s",
+				keystoreSecret.Name, ownerRef.Kind, ownerRef.Name, ownerRef.UID)
+		}
+		return controllerutil.SetControllerReference(i, keystoreSecret, s.scheme)
+	})
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+	return nil, nil
+}
+
+func (s secretResource) reconcileTruststoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsClientCertEnabled() {
+		return nil, nil
+	}
+
+	trustSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      i.GetTruststoreSecretName(),
+			Namespace: i.Namespace,
+		},
+	}
+
+	if trustSecret.Name == "" {
+		// It's not possible for client cert to be configured if no truststore secret is provided by the user
+		// as no certificates are available to be added to the server's truststore, resulting in no client's
+		// being able to authenticate with the server.
+		return &reconcile.Result{}, fmt.Errorf("Field 'ClientCertSecretName' must be provided for '%s' or '%s' to be configured",
+			ispnv1.ClientCertAuthenticate, ispnv1.ClientCertValidate)
+	}
+
+	if result, err := kube.LookupResource(trustSecret.Name, i.Namespace, trustSecret, s.client, s.log); result != nil {
+		return result, err
+	}
+
+	_, err := kube.CreateOrPatch(ctx, s.client, trustSecret, func() error {
+		if trustSecret.CreationTimestamp.IsZero() {
+			return errors.NewNotFound(corev1.Resource("secret"), i.GetTruststoreSecretName())
+		}
+
+		_, truststoreExists := trustSecret.Data[consts.EncryptTruststoreKey]
+		if truststoreExists {
+			if _, ok := trustSecret.Data[consts.EncryptTruststorePasswordKey]; !ok {
+				return fmt.Errorf("The '%s' key must be provided when configuring an existing Truststore", consts.EncryptTruststorePasswordKey)
+			}
+		} else {
+			caPem := trustSecret.Data[EncryptClientCAName]
+			if trustSecret.Data == nil {
+				trustSecret.Data = map[string][]byte{
+					consts.EncryptTruststorePasswordKey: []byte(EncryptTruststorePassword),
+				}
+			} else {
+				if _, passwordProvided := trustSecret.Data[consts.EncryptTruststorePasswordKey]; !passwordProvided {
+					trustSecret.Data[consts.EncryptTruststorePasswordKey] = []byte(EncryptTruststorePassword)
+				}
+			}
+
+			certs := [][]byte{caPem}
+			for certKey := range trustSecret.Data {
+				if strings.HasPrefix(certKey, EncryptClientCertPrefix) {
+					certs = append(certs, trustSecret.Data[certKey])
+				}
+			}
+			password := string(trustSecret.Data[consts.EncryptTruststorePasswordKey])
+			truststore, err := security.GenerateTruststore(certs, password)
+			if err != nil {
+				return err
+			}
+			trustSecret.Data[consts.EncryptTruststoreKey] = truststore
+			return nil
+		}
+		return controllerutil.SetControllerReference(i, trustSecret, s.scheme)
```

Comment:
Do we need to add the same validation before `SetControllerReference` as for KeystoreSecret?